spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/datamining/RowTransformer.scala

```diff
@@ -0,0 +1,324 @@
+  override val schemaKey: String,
+  override val indices: Seq[Int] = Seq.empty,
+  override val fieldNames: Seq[String] = Seq.empty
+)(implicit ev: TensorNumeric[T]) extends RowTransformSchema {
+
+  override def transform(input: Seq[Any], fields: Seq[StructField]): Tensor[NumericWildcard] = {
+    val tensor = Tensor[T](input.length)
+    var i = 0
+    while (i < input.length) {
+      val value = fields(i).dataType match {
+        // TODO: support VectorUDT
+        case _: DoubleType => ev.fromType(input(i).asInstanceOf[Double])
+        case _: FloatType => ev.fromType(input(i).asInstanceOf[Float])
+        case _: ShortType => ev.fromType(input(i).asInstanceOf[Short])
+        case _: IntegerType => ev.fromType(input(i).asInstanceOf[Int])
+        case _: LongType => ev.fromType(input(i).asInstanceOf[Long])
+        case tpe => throw new IllegalArgumentException(s"Found unSupported DataType($tpe)!")
+      }
+      tensor.setValue(i + 1, value)
+      i += 1
+    }
+    tensor.asInstanceOf[Tensor[NumericWildcard]]
+  }
+
+}
+
+object ColsToNumeric {
+
+  /**
+   * Build a `ColsToNumeric` which transforms `all columns` of Row.
+   *
+   * @param schemaKey key of the schema
+   * @tparam T the type of result `Tensor`
+   */
+  def apply[@specialized(Float, Double) T: ClassTag](schemaKey: String
+  )(implicit ev: TensorNumeric[T]): ColsToNumeric[T] = {
+    new ColsToNumeric[T](schemaKey)
+  }
+
+  /**
+   * Build a `ColsToNumeric` which transforms `selected columns` of Row.
+   *
+   * @param schemaKey key of the schema
+   * @param fieldNames field names of `selected columns`
+   * @tparam T the type of result `Tensor`
+   */
+  def apply[@specialized(Float, Double) T: ClassTag](
+    schemaKey: String,
+    fieldNames: Seq[String]
+  )(implicit ev: TensorNumeric[T]): ColsToNumeric[T] = {
+    new ColsToNumeric[T](schemaKey, Seq.empty, fieldNames)
+  }
+
+}
+
+/**
+ * A schema which specialize on transforming `single column` to size(1) `Tensor`.
+ * Types of `selected columns` will be identified according to their `DataType`.
+ * And type conversions will be done automatically from `DataType` to `TensorDataType`.
+ *
+ * @param schemaKey key of the schema
+ * @param index index of selected column, overridden by non empty `fieldName`
+ * @param fieldName field name of selected column, default is empty
+ */
+class ColToTensor(
+  override val schemaKey: String,
+  index: Int,
+  fieldName: String = ""
+) extends RowTransformSchema {
+
+  override val indices: Seq[Int] = Seq(index)
+
+  override val fieldNames: Seq[String] = if (fieldName.isEmpty) Seq.empty else Seq(fieldName)
+
+  override def transform(input: Seq[Any], fields: Seq[StructField]): Tensor[NumericWildcard] = {
+    val (value, tpe) = input.head -> fields.head.dataType
+    val tensor = tpe match {
+      // TODO: support VectorUDT
+      case _: BooleanType => Tensor[Boolean](1).setValue(1, value.asInstanceOf[Boolean])
```

Comment:
Tensor.scala\[Boolean\](value)?

Reply:
I think the _Size_ of output _Tensor_ is necessary for subsequent batch-like actions(such as _SampleToMiniBatch_)  :-)

Reply:
ok